commands/translation_stat/translation_stat.py

```diff
@@ -0,0 +1,223 @@
+
+    async def track_handler(
+        interaction: discord.Interaction, locale: str, pr_no: int = None
+    ):
+        await interaction.response.defer()
+        if locale != core.main_lang:
+            core.require_locale(core.main_lang)
+        core.require_locale(locale)
+        ref = core.trans_db[locale]
+        all_progresses = pr_files = master_files = None
+        with ref.mutex:
+            if pr_no:
+                if ref.pr_no and ref.pr_no == pr_no:
+                    await interaction.followup.send(
+                        embed=discord.Embed(description=f"❌  **{locale} is already tracking PR {pr_no}**", colour=discord.Colour.red())
+                    )
+                    return
+                core.shift2pr(locale, pr_no)
+            else:
+                if not ref.pr_no:
+                    await interaction.followup.send(
+                        embed=discord.Embed(description=f"❌  **{locale} is already tracking Master**", colour=discord.Colour.red())
+                    )
+                    return
+                core.shift2master(locale)
+
+            all_progresses = get_file_progresses(locale, ref.all_files())
+            tabulate_list = monospace_filename_table(tabulate(
+                progresses_to_str(all_progresses), tablefmt="plain"
+            ).splitlines(False))
+            pr_files = match_table_rows(tabulate_list, ref.pr_files)
+            master_files = match_table_rows(tabulate_list, ref.owned_files)
+
+        def concat_progress_str(file_progresses: dict[str, str]) -> str:
+            nonlocal all_progresses
+            ret = str()
+            for key, value in file_progresses.items():
+                ret += f"{value}\n"
+                ret += f"{all_progresses[key].to_progress_str()}\n"
+            return ret
+
+        embed = discord.Embed(
+            title=f"Source for Language {locale}",
+            description=f"{locale} is now tracking {'Master' if not pr_no else f'PR {pr_no}'}",
+            colour=discord.Colour.green()
+        )
+        embed.add_field(
+            name="Files provided by Pull Request",
+            value=concat_progress_str(pr_files) if len(pr_files) else "None",
+            inline=False,
+        )
+        embed.add_field(
+            name="Files provided by Master",
+            value=concat_progress_str(master_files)
+            if len(master_files)
+            else "None",
+        )
+        embed.set_footer(
+            text=f"See /trans_stat {locale} for more information"
+        )
+        await interaction.followup.send(content=None, embed=embed)
+
+    @track_group.command(
+        description="Set language tracking source to Pull Request",
+    )
+    @app_commands.choices(
+        lang=[Choice(name=lang, value=lang) for lang in core.trans_db.keys()]
+    )
+    @app_commands.describe(pr_no="Pull request number")
+    async def pr(
+        interaction: discord.Interaction, lang: Choice[str], pr_no: int
+    ):
```

Comment:
Hmm, would it be better to track branch name instead of PR number? (ex. `bot-i18n-commits` is where translations made through the bot all go to and we make PRs for it periodically)